examples/asa/scripts/2-gold-asc.js

```diff
@@ -1,35 +1,41 @@
 const { balanceOf, signTransactions } = require("@algo-builder/algob");
 const { types, getSuggestedParams } = require("@algo-builder/web");
-const { mkParam } = require("./transfer/common");
+const { mkParam, tryExecuteTx } = require("./transfer/common");
 const { makeAssetTransferTxnWithSuggestedParams } = require("algosdk");
 
 async function run(runtimeEnv, deployer) {
 	const masterAccount = deployer.accountsByName.get("master-account");
 	const goldOwner = deployer.accountsByName.get("alice");
 
-	await deployer.executeTx(
+	await tryExecuteTx(
+		deployer,
 		mkParam(masterAccount, goldOwner.addr, 200e6, { note: "funding account" })
 	);
 
 	// save Smart Signature by name & fund the account
-	const ascInfoContract = await deployer.mkContractLsig(
-		"Gold_C_Lsig",
-		"2-gold-contract-asc.teal",
-		{}
-	);
+	const ascInfoContract = await deployer
+		.mkContractLsig("Gold_C_Lsig", "2-gold-contract-asc.teal", {})
+		.catch((error) => {
```

Comment:
why do we catch it and imidietly throw it again here?

Reply:
because else the tests won't fail in case there is an error, it might just log (depending upon what we pass in catch)

Reply:
hm I wa thinking that if the error is thrown inside of `mkContractLsig` and we do not catch it it will behave exactly the same

Reply:
It should but we don't throw error in `mkContractLsig `